### ISSUE TITLE
WX 4100, Fan Curve not accurate

### ISSUE THREAD

**Issue by knuxyl** [ilya-zlobintsev/LACT#311]
### Checklist

- [X] The problem is not listed in the [hardware support matrix](https://github.com/ilya-zlobintsev/LACT?tab=readme-ov-file#hardware-support) as a known limitation
- [X] I've included a [debug snapshot](https://github.com/ilya-zlobintsev/LACT?tab=readme-ov-file#reporting-issues) if the issue is related to hardware configuration

### Bug description

Fan is running about 20% faster than what the curve is set to. I've been watching it for 30 minutes and it never slows down. I tried changing some settings but it seems to give a consistent ~20% increase of set values to fan speed.

I was originally using amdgpufan service to control speeds through a configuration file and it worked as expected. I disabled the service, installed this, enabled the daemon, restarted, and have been experiencing this issue since.

LACT does control the fan successfully, just not correctly.

![Screenshot from 2024-04-23 09-40-43](https://github.com/ilya-zlobintsev/LACT/assets/19507326/d58b3040-4c5e-41da-85b7-22c41b41ded7)

### System info

[LACT-sysfs-snapshot-20240423-072913.tar.gz](https://github.com/ilya-zlobintsev/LACT/files/15078576/LACT-sysfs-snapshot-20240423-072913.tar.gz)

```markdown
- LACT version: 0.5.4-release (commit 6dc3520)
- GPU model: AMD Radeon Pro WX 4100 (0x1002:0x67E3:0x0)
- Kernel version: 6.6.15-amd64
- Distribution: Debian Testing
```

**Comment by ilya-zlobintsev** [comment]
Did you take the screenshot with a curve at a different time than the snapshot? Because the snapshot has a different curve in the config (`/etc/lact/config.yaml`):
```
      curve:
        30: 0.0
        35: 0.2
        40: 0.4
        50: 0.6006192
        60: 0.7987616
        70: 1.0
```
Also, does the static speed setting work correctly?

**Comment by knuxyl** [comment]
Yeah it's from a different time. That config matches what it was set at the time. Ill have to check static speed later.

**Comment by Slater91** [comment]
I see the same with a WX 3100, except that there is a 15% discrepancy instead of 20%.

**Comment by sladiri** [comment]
I seem to have the same issue, using an RX6600XT.

**Comment by RealOjive** [comment]
I might have the same or similar issue on 5700XT. When I set all curves/temps to 0% speed, its silent. If I set curve 1 (40C) to 2%, the mid fan of three fans on my 5700XT Nitro goes loud. 

I manually verified that its the mid fan that is loud by physically stopping fan 1 and 3 and its the 2nd (mid) fan that is loud. It doesn't matter if its 2% or 10% speed.

**Comment by knuxyl** [comment]
I got the numbers for my other GPU, Pro W6600, and I have 70% fan at 50C and 80% fan at 60C. GPU has been idling at 54C and it is pushing 92% fan. This hasn't been working correctly for any of my GPUs since I first installed this but I just deal with it. I would prefer if it would set the correct fan speed but my cards get so damn hot anyways it doesn't matter much. But this is definitely still an issue.

**Comment by ilya-zlobintsev** [comment]
LACT sets the fan speed value as a percentage (or more specifically, a 0-255 PWM value) directly to the GPU. How exactly the GPU treats this value is not controlled by LACT. Based on the high RPM value I assume this is a small fan on a professional card, which likely has a very non-linear PWM curve making it set higher-than-configured speeds.

You can confirm this by trying to set the fan speed manually (set fan control to automatic in LACT first so it doesn't overwrite):
```
echo 1 | sudo tee /sys/class/drm/card*/device/hwmon/hwmon*/pwm1_enable # Enable manual
echo 128 | sudo tee /sys/class/drm/card*/device/hwmon/hwmon*/pwm1 # Set pwm value to 128 - this should be 50%
```
